packages/oracle/src/connection-manager.ts

```
@@ -0,0 +1,181 @@
+}
+
+export interface OracleConnectionOptions extends oracledb.ConnectionAttributes {
+  database?: string;
+
+  host?: string;
+
+  oracleOptions?: object;
+
+  port?: number | string;
+}
+
+export class OracleConnectionManager extends AbstractConnectionManager<
+  OracleDialect,
+  OracleConnection
+> {
+  lib: typeof oracledb;
+  constructor(dialect: OracleDialect) {
+    super(dialect);
+    this.lib = oracledb;
+  }
+
+  buildConnectString(config: ConnectionOptions<OracleDialect>) {
+    if (!config.host || config.host.length === 0) {
+      return config.database;
+    }
+
+    let connectString = config.host;
+    if (config.port) {
+      connectString += `:${config.port}`;
+    } else {
+      connectString += ':1521';
+    }
+
+    if (config.database && config.database.length > 0) {
+      connectString += `/${config.database}`;
+    }
+
+    return connectString;
+  }
+
+  /**
+   * Method for initializing the lib
+   *
+   * @param config
+   */
+  extendLib(config: ConnectionOptions<OracleDialect>) {
+    if (config.oracleOptions) {
+      if ('maxRows' in config.oracleOptions) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-expect-error add maxRow
+        oracledb.maxRows = config.oracleOptions.maxRows;
+      }
+
+      if ('fetchAsString' in config.oracleOptions) {
+        // @ts-expect-error -- add fetchAsString
+        oracledb.fetchAsString = config.oracleOptions.fetchAsString;
+      }
+    }
+
+    oracledb.fetchAsString = [oracledb.CLOB];
+
+    // Retrieve BLOB always as Buffer.
+    oracledb.fetchAsBuffer = [oracledb.BLOB];
+  }
+
+  async connect(config: ConnectionOptions<OracleDialect>): Promise<OracleConnection> {
+    this.extendLib(config);
+    this.lib = oracledb;
+    const connectionConfig: OracleConnectionOptions = {
+      username: config.username,
+      password: config.password,
+      connectString: this.buildConnectString(config),
```

`connectString` is marked as being allowed as a configuration option, so you should throw an error if `database`, `host` and `port` are provided at the same time as `connectString`

---

Added error in buildConnectString().